tests/unit/test_utils.py

```diff
@@ -203,3 +210,213 @@ def test_can_silence_io_error_from_pager(self):
                     pass
         except IOError:
             self.fail('Should not raise IOError')
+
+
+class TestInstanceMetadataRegionFetcher(unittest.TestCase):
```

Comment:
Talked offline about this as well, but it would be great to get a functional test in for this.